src/main/kotlin/me/dkim19375/dkimgradle/util/CommonScripts.kt

```diff
@@ -26,25 +26,61 @@ package me.dkim19375.dkimgradle.util
 
 import me.dkim19375.dkimgradle.annotation.API
 import me.dkim19375.dkimgradle.delegate.TaskRegisterDelegate
+import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.Task
-import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.compile.JavaCompile
 import org.gradle.kotlin.dsl.named
+import org.gradle.language.jvm.tasks.ProcessResources
 import java.io.File
 
+/**
+ * Sets up the project with the specified [group] and [version]
+ *
+ * Adds the text encoding, replacements, and build shadow task (if the Shadow plugin is applied)
+ */
+fun Project.setup(group: String, version: String) {
+    this.group = group
+    this.version = version
+    // Tasks
+    addTextEncodingTask()
+    addReplacementsTask()
+    if (hasShadowPlugin()) addBuildShadowTask()
+}
+
+/**
+ * Checks if the Shadow plugin is applied
+ */
+fun Project.hasShadowPlugin(): Boolean = plugins.hasPlugin("com.github.johnrengelman.shadow")
+
+/**
+ * Adds the task that makes `gradle build` run `gradle shadowJar`
+ */
+fun Project.addBuildShadowTask() {
+    check(hasShadowPlugin()) { "Shadow plugin is not applied!" }
+    tasks.named<DefaultTask>("build") { dependsOn("shadowJar") }
+}
+
+/**
+ * Adds the task to set the text encoding to UTF-8
+ */
+fun Project.addTextEncodingTask() {
```

Comment:
I know it's kinda inconsistent but can you change this to like `setTextEncoding(encoding: String = "UTF-8")` since it's not enabling any feature or anything (compared to `addReplacementsTask` where I'm "enabling" the replacements feature)